subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiSpecification.groovy

```diff
@@ -131,11 +132,14 @@ abstract class ToolingApiSpecification extends Specification {
     }
 
     TestFile getBuildFileKts() {
+        validateKotlinCompatibility()
         file("build.gradle.kts")
     }
 
-    TestFile getBuildKotlinFile() {
-        getBuildFileKts()
+    private validateKotlinCompatibility() {
+        if (targetGradleDistribution && !targetGradleDistribution.supportsKotlinScript) {
+            throw new GradleException("The current Gradle target version ($targetGradleDistribution.version) does not support execution of Kotlin build scripts.")
```

Comment:
I think a simple `RuntimeException` is enough here. `GradleException` is should be thrown by production code only. I guess this is why you needed to declare the new dependency, which should not be necessary.